flink-runtime/src/main/java/org/apache/flink/runtime/rpc/RpcSystemUtils.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.rpc;
+
+import org.apache.flink.configuration.Configuration;
+
+import javax.annotation.Nullable;
+
+import java.util.Optional;
+
+/** Utils related to the {@link RpcSystem}. */
+public final class RpcSystemUtils {
+
+    /**
+     * Convenient shortcut for constructing a remote RPC Service that takes care of checking for
+     * null and empty optionals.
+     *
+     * @see RpcSystem#remoteServiceBuilder(Configuration, String, String)
+     */
+    public static RpcService createRemoteRpcService(
```

Comment:
Does it make sense to move this method into the `RpcUtils` class? The downside of introducing a new `RpcSystemUtils` class is that helper methods are harder to find.

Reply:
yes that would make sense; this separation doesn't really provide a benefit.